### PR TITLE
refactor(dfx): unify L2 phase buffer switching

### LINKED ISSUE #1385: [Code Health] Unify L2 swimlane buffer-switch wrappers: dedup phase free-queue pop onto engine helpers
### Category

Technical Debt (cleanup, refactor)

### Component

Platform (a2a3 / a2a3sim) — cross-arch DFX collector under `src/common/platform/`

### Description

The L2 swimlane DFX collector routes 4 buffer kinds (`AicpuTask`, `AicpuSchedPhase`, `AicpuOrchPhase`, `AicoreTask`) through the shared device engine, but the **producer-side buffer-switch wrappers are asymmetric** across three families:

| Family | Surface wrapper | Producer model | Engine usage |
| --- | --- | --- | --- |
| `AicpuTask` | `switch_records_buffer` → `L2SwimlaneTaskEngine::switch_buffer` / `pop_free` | AICPU produces & recycles | **fully** uses engine helpers |
| `AicpuSchedPhase` / `AicpuOrchPhase` | `switch_phase_buffer_kind` + `acquire_phase_slot` | AICPU produces & recycles (same `L2SwimlaneAicpuTaskPool` struct) | uses `enqueue_ready` only; **hand-rolls the free-queue pop** |
| `AicoreTask` | `flush_aicore_pending_enqueue` / aicore rotate | **AICore** produces, AICPU only publishes | `enqueue_ready` + stash-retry (publish-only) |

The load-bearing part — the push/pop **gate** (`enqueue_ready` / `wait_for_free_queue_entry`, incl. the backpressure freeze barriers) — is already unified across all 4 kinds. The concern is the wrappers *around* the gate:

1. **Incidental duplication (the real cost).** `switch_phase_buffer_kind` and `acquire_phase_slot` re-implement, by hand, exactly what the engine `pop_free` / `switch_buffer` already do: `wait_for_free_queue_entry` → `buffer_ptrs[head % PLATFORM_PROF_SLOT_COUNT]` → `free_queue.head = head + 1` → set `current_buf_ptr` / `current_buf_seq`. This logic now lives in two places (once in the engine, once hand-rolled for phase) and can **drift** — a change to the seq/head bookkeeping in one spot silently diverges from the other. Notably the phase pool is the **same `L2SwimlaneAicpuTaskPool` struct** as task, so this duplication is not forced by the data model.
2. **Inconsistent naming.** The three families have ad-hoc names (`switch_records_buffer` / `switch_phase_buffer_kind` / `flush_aicore_pending_enqueue`), so a reader cannot tell at a glance which are the same operation and which differ.

Scope note: `AicoreTask` is a genuinely different (cross-processor, publish-only) operation and should **not** be forced under a common `switch()` API — sharing only the `enqueue_ready` gate, as it does today, is the correct boundary. Full "4 kinds → 1 switch" is explicitly a non-goal (it would reintroduce the runtime type-switch the template-Module design deliberately avoids).

This is pre-existing structure, **not** introduced by #1313 — filing separately so the PR stays scoped.

### Location

- `src/common/platform/shared/aicpu/l2_swimlane_collector_aicpu.cpp`
  - `enqueue_ready_buffer` (shared push wrapper) — L221
  - `switch_records_buffer` (AicpuTask) — L368
  - `flush_aicore_pending_enqueue` (AicoreTask, publish-only) — L394
  - `switch_phase_buffer_kind` (hand-rolled pop) — L821
  - `acquire_phase_slot` (hand-rolled pop) — L880
- Engine helpers the phase path should reuse: `src/common/platform/include/aicpu/profiler_device_engine.h` — `pop_free` (L150), `switch_buffer` (L180)

### Proposed Fix

- **Tier 2 (main value): dedup.** Route the phase (`sched` / `orch`) free-queue pop through the engine `pop_free` / `switch_buffer` instead of hand-rolling it, so there is a single source of truth for free-queue head/seq bookkeeping. Obstacle to resolve: the engine is `DeviceProfilerEngine<Module>` and phase records use a different buffer type — parameterize the Module/Context over the record-buffer type (or add a phase Module instantiation).
- **Naming: unify** the three families under a consistent scheme that makes the producer model explicit (e.g. `l2_task_switch` / `l2_phase_switch` / `l2_aicore_publish`) rather than hiding the difference.
- **Non-goal:** collapsing all 4 kinds into one `switch(kind)` dispatcher; keep `AicoreTask` separate.

Related: #1313 (the DFX backpressure PR that touches this file), #997 (parent DFX backpressure feature).

### Priority

Low (no impact today, good to fix eventually)

### PR DESCRIPTION
**Pull request by vegetabledoww**
## Summary

- Reuse `DeviceProfilerEngine` for sched/orch phase buffer switching and recovery.
- Centralize free-queue ownership, head updates, and sequence bookkeeping.
- Keep phase startup acquisition non-blocking through `try_pop_free()`.
- Clarify task/phase/AICore wrapper naming.
- Add unit tests for buffer switching, replacement failure, recovery, and empty startup queues.
- Preserve the reserved sequence number when recovering after a replacement-pop failure.

Fixes #1385

## Validation

- `test_profiler_device_engine`: passed
- a2a3 AICPU kernel: built successfully
- a5 AICPU kernel: built successfully
- pre-commit checks: passed

### CODE REVIEW

File: src/common/platform/shared/aicpu/l2_swimlane_collector_aicpu.cpp [hunk 1]
```diff
@@ -882,29 +884,19 @@ static Record *acquire_phase_slot(
...
-            state->head.current_buf_ptr = buf_ptr;
-            state->head.current_buf_seq += 1;
-            wmb();
-            buf = reinterpret_cast<Buffer *>(buf_ptr);
-            buf->count = 0;
-            *current_buf_out = buf;
+        auto ctx = l2_buffer_context(thread_idx, pool_idx, kind, current_buf_out);
+        buf = profiling_device::DeviceProfilerEngine<L2SwimlaneDeviceModule<Buffer>>::pop_free(
+            ctx, state, state->head.current_buf_seq + 1
+        );
```
[the remark this answers is not in the document]
Comment (vegetabledoww) [reply src/common/platform/shared/aicpu/l2_swimlane_collector_aicpu.cpp:890]: Fixed in 73e8ccd4. `switch_buffer()` already reserves the next sequence before attempting the replacement pop, so the recovery path now passes `state->head.current_buf_seq` directly instead of incrementing it again. The recovery test was updated to assert that sequence 8 is preserved. Validated with `test_profiler_device_engine` and both a2a3/a5 AICPU kernel builds.

[the remark this answers is not in the document]
Comment (vegetabledoww) [reply src/common/platform/shared/aicpu/l2_swimlane_collector_aicpu.cpp:890]: Fixed in 73e8ccd4. `switch_buffer()` already reserves the next sequence before attempting the replacement pop, so the recovery path now passes `state->head.current_buf_seq` directly instead of incrementing it again. The recovery test was updated to assert that sequence 8 is preserved. Validated with `test_profiler_device_engine` and both a2a3/a5 AICPU kernel builds.
